tests/fourmolu.nix

```diff
@@ -0,0 +1,29 @@
+{ runCommand, fourmolu, src, lib }:
+
+let
+  # Remove everything except Haskell sources, fourmolu configuration, and cabal files.
+  src' = lib.cleanSourceWith {
+    inherit src;
+
+    filter = with lib; name: type:
+      let baseName = baseNameOf (toString name); in (
+        (type == "regular" && hasSuffix ".hs" baseName) ||
+        (type == "regular" && hasSuffix ".yaml" baseName) ||
+        (type == "regular" && hasSuffix ".cabal" baseName) ||
+        (type == "directory")
+      );
+  };
+in
+
+runCommand "fourmolu-check" { buildInputs = [ fourmolu ]; } ''
+  set +e
+  cd ${src'}
+  fourmolu --mode check .
+  EXIT_CODE=$?
+
+  if [[ "$EXIT_CODE" -eq "0" ]]; then
```

Comment:
If the exit code indicates success, then we write something into `"$out"`. It doesn't matter what it is, because without that, nix considers it a failure

Reply:
But you could get rid of this `if`, right? Leaving only `echo`.  Since if you exit with non-0, it will fail anyway

Reply:
Yes, that's true. I've now removed that section.